### PR TITLE
Add support for propagating the X-Request-Id header

### PR DESCRIPTION
**Pull request by andrenpaes**
Keeps backward compatibility to the old "X-Killbill-Request-Id-Req" header.

### CODE REVIEW

File: profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java [hunk 1]
```diff
@@ -32,12 +34,13 @@
 @Singleton
 public class RequestDataFilter implements ContainerRequestFilter, ContainerResponseFilter {
 
+    private static final String REQUEST_ID_HEADER = "X-Request-Id";
 
-    private static final String REQUEST_ID_HEADER_REQ = "X-Killbill-Request-Id-Req";
+    private static final String LEGACY_REQUEST_ID_HEADER = "X-Killbill-Request-Id-Req";
```
Comment (pierre) [inline profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java]: :+1: for backward compatibility.

Nit: maybe add a `@Deprecated`?

**Comment by pierre** [comment]
Since you are touching that code, maybe you could also address the second part of https://github.com/killbill/killbill/issues/409, i.e. set the `userToken` of the `CallContext` to the `requestId`, if set (see [Context.java](https://github.com/killbill/killbill/blob/master/jaxrs/src/main/java/org/killbill/billing/jaxrs/util/Context.java#L51-L52)). This would make the request id flow to the database (audit logs, bus events, notifications, etc.).

Unclear how big of a change it is, we could also do it in a another PR. The first part of https://github.com/killbill/killbill/issues/409 is definitively much bigger though.

File: profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java [hunk 2]
```diff
@@ -32,12 +34,13 @@
 @Singleton
 public class RequestDataFilter implements ContainerRequestFilter, ContainerResponseFilter {
 
+    private static final String REQUEST_ID_HEADER = "X-Request-Id";
```
Comment (sbrossie) [inline profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java]: Should it be `X-Request-ID` or `X-Request-Id`. Googling shows me a lot of `X-Request-ID`. Unclear...

Comment (andrenpaes) [inline profiles/killbill/src/main/java/org/killbill/billing/server/filters/RequestDataFilter.java]: For future reference, according to the HTTP spec, headers are case insensitive, so either way should work.

**Pull request merged**